### ISSUE TITLE
OpenGL wrapper not loading

### ISSUE THREAD

**Issue by casasfernando** [kjliew/qemu-3dfx#75]
I'm testing qemu-3dfx (6.2.0) running a Windows 98 SE guest on Ubuntu 20.04 host with Intel gen 11th CPU and Iris Xe iGPU.
I tested a some Glide games so far with great success.
But now I'm trying to test Soldier of Fortune using the opengl wrapper and I can't get it to work.

I got wglgears to start with the most basic testing. With Windows stock opengl32.dll, I get the gears working, with very low FPS which I think is expected. But if I copy the opengl32.dll I built when compiled qemu, the glide wrappers and openglide, to the same folder of wglgears and try to run it I get the following error and wglgears exit right away:

![image](https://user-images.githubusercontent.com/15861819/201501303-a9a75499-71b2-4146-b951-dd0b0cc13ae4.png)

In qemu output in the console I don't get any message at all (unlike with glide which shows when the dll is loaded, etc) so I'm almost sure that the wrapper is not even trying to load. This is my qemu command line:

```
LD_LIBRARY_PATH=$PWD/libopenglide ./qemu-system-i386 -m 1024M -nodefaults -rtc base=localtime -display sdl -M pc,accel=kvm,hpet=off,usb=off -cpu host -device cirrus-vga -device ac97 -netdev user,id=net0 -device e1000,rombar=0,netdev=net0     -drive id=win98se,if=none,file=en_win98se_retail.qcow2,driver=qcow2,aio=threads,cache=none -device ide-hd,drive=win98se     -cdrom post-install.iso -boot c
```

While building the wrapper I didn't get any errors:

```
checking for make... /usr/bin/make
checking for sed... /bin/sed
checking for xxd... /usr/bin/xxd
checking for gendef... /usr/bin/gendef
checking for shasum... /usr/bin/shasum
checking for git... /usr/bin/git
  Linux cross build
  CROSS=i686-w64-mingw32-
  CC=gcc
-- DOS32 OVL not supported --
-- DJGPP DXE not supported --
conf_wrapper: creating Makefile
  CC  fxhook.o fxlib9x.o fxlibnt.o
  CC  szdata.o
  CC  wrapgl32.o
  REV stamp.o "d2ce64a-"
  CFLAGS  -I../../../qemu-1/hw/mesa -I../../fxlib -Wall -Werror -msse2 -O3 -flto -fomit-frame-pointer
  LDFLAGS -static-libgcc -Wl,--disable-auto-image-base,--no-seh,--dynamicbase,--nxcompat
 * [opengl32.dll] Found PE image
  LD  opengl32.dll
```

And as I mentioned before glide wrappers are working great.
Any ideas on what I'm missing?

Thanks.

**Comment by kjliew** [comment]
Search the closed issues for answers or make a small donation to support the project for QEMU binary package that includes MESA GL pass-through for the host platform of choice. Ubuntu 20.04 is a popular & well supported platform among project donors. Details at https://github.com/kjliew/qemu-3dfx#donation

**Comment by casasfernando** [comment]
After a few months I decided to go back and try to solve this problem. I found the cause of the issue and decided to share it here in case anyone hits the same problem in the future.

I'm running on Ubuntu 22.04 and looking at the source code I realized that I was missing symlink in my system: **libGL.so -> /lib/x86_64-linux-gnu/libGL.so.1.7.0**
After creating the missing symlink manually (didn't want to install libgl-dev package) the problem was solved and now the wrapper is loading as expected.
